velox/common/encode/EncoderUtils.h

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <exception>
+#include <map>
+#include <string>
+
+#include <folly/Range.h>
+
+namespace facebook::velox::encoding {
+
+class EncoderException : public std::exception {
+ public:
+  explicit EncoderException(const char* msg) : msg_(msg) {}
+  const char* what() const noexcept override {
+    return msg_;
+  }
+
+ protected:
+  const char* msg_;
+};
+
+constexpr size_t CHARSET_SIZE = 64;
+constexpr size_t REVERSE_INDEX_SIZE = 256;
+
+using Charset = std::array<char, CHARSET_SIZE>;
```

Comment:
What do these represent and what is the relation between them ? Please can you add a comment about it.

Reply:
Added the comments